src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,193 @@
+        }
+        String description = fullDescription.substring(0, sepIndex - 1);
+        String dateTime = fullDescription.substring(sepIndex + 4);
+        String[] dateTaskArray = dateTime.split(" ");
+        if (dateTaskArray.length == 2) {
+            System.out.println(dateTime);
+            tasks.add(new Deadline(description, dateTime, FORMAT_TIME, true, false));
+        } else {
+            tasks.add(new Deadline(description, dateTime, FORMAT_NO_TIME, false, false));
+        }
+        return getLatestTaskString();
+    }
+
+    /**
+     * Adds an duke.task.Event task.
+     *
+     * @param fullDescription String that contains the description and time of the task.
+     * @return Response to an event being added.
+     */
+    public String addEvent(String fullDescription) throws DukeException {
+        int sepIndex = fullDescription.indexOf("/at");
+        if (fullDescription.charAt(sepIndex + 3) != ' ' || sepIndex == -1 || sepIndex == 0 ||
+                fullDescription.charAt(sepIndex - 1) != ' ') {
+            throw new DukeException("☹ OOPS!!! Please input with the correct format e.g. event read books" +
+                    " /at 2021-09-08 09:30 (yyyy-mm-dd hh:mm, where time is optional)");
+        }
+        String description = fullDescription.substring(0, sepIndex - 1);
+        String dateTime = fullDescription.substring(sepIndex + 4);
+        String[] dateTaskArray = dateTime.split(" ");
+        if (dateTaskArray.length == 2) {
+            tasks.add(new Event(description, dateTime, FORMAT_TIME, true, false));
+        } else {
+            tasks.add(new Event(description, dateTime, FORMAT_NO_TIME, false, false));
+        }
+        return getLatestTaskString();
+    }
+
+    /**
+     * Prints information about the latest task that was just added.
+     *
+     * @return String representing the latest task that was added.
+     */
+    public String getLatestTaskString() {
+        String result = "Got it. I've added this task:\n";
+        result += "   " + tasks.get(tasks.size() - 1) + "\n";
+        result += "Now you have " + tasks.size() + " task(s) in the list.";
+        System.out.println("Got it. I've added this task:");
+        System.out.println("   " + tasks.get(tasks.size() - 1));
+        System.out.println("Now you have " + tasks.size() + " task(s) in the list.");
+        return result;
+    }
+
+    /**
+     * Deletes a specific task.
+     *
+     * @param taskId ID of the task to be deleted.
+     * @return Response to the task being deleted.
+     */
+    public String deleteTask(int taskId) throws DukeException {
+        if (taskId < 1 || taskId > tasks.size()) {
+            throw new DukeException("☹ OOPS!!! Please provide a task ID that exists.");
+        }
+        String output = "Noted. I have removed this task:\n";
+        output += "   " + tasks.get((taskId - 1)) + "\n";
+        System.out.println("Noted. I have removed this task:");
```

Comment:
Console outputs could be abstracted out into a separate class, so that it would be easy to shift from CLI to GUI for example